### ISSUE TITLE
null is not an object (evaluating 'HMSManager.build')

### ISSUE THREAD

**Issue by charles-mersereau** [100mslive/react-native-hms#718]
When I create a new project and follow the tutorial steps on the [100ms React Native guide](https://www.100ms.live/docs/react-native/v2/guides/quickstart), I am consistently getting this error.

Reproduction steps:

1. Create a new React Native project (I have tried expo init and create-react-native-app)
2. Navigate into project and install dependency with `npm install @100mslive/react-native-hms --save`
3. Initialize the sdk with `import HmsManager from '@100mslive/react-native-hms';` and calling `HmsManager.build();`
4. Run the project using `npm start` or `expo start`

Here's my initialization code, in a project using reacts tabbed template:

```
import { useState, useEffect } from 'react';

import { StatusBar } from 'expo-status-bar';
import { SafeAreaProvider } from 'react-native-safe-area-context';

import useCachedResources from './hooks/useCachedResources';
import useColorScheme from './hooks/useColorScheme';
import Navigation from './navigation';

import HmsManager from '@100mslive/react-native-hms';


export default function App() {
  const isLoadingComplete = useCachedResources();
  const colorScheme = useColorScheme();
  const [hmsInstance, setHmsInstance] = useState({});
  const [isHmsInstanceInitialized, setHmsInstanceInitialized] = useState(false);

  useEffect(() => {
    if (isHmsInstanceInitialized)
    {
      return;
    }

    console.log("manager: " + HmsManager);
      HmsManager.build()
        .then((instance) => {
          console.log("hms instance: " + instance);
          setHmsInstance(instance);
          setHmsInstanceInitialized(true);
        })
        .catch((console.error));

  }, []);


  if (!isLoadingComplete) {
    return null;
  } else {
    return (
      
      <SafeAreaProvider>
        <Navigation colorScheme={colorScheme} />
        <StatusBar />
      </SafeAreaProvider>
    );
  }
}
```

When opening the app on an Android device (Google Pixel 5, Samsung Note 20), I get the following error:

`null is not an object (evaluating 'HMSManager.build')
at node_modules\@100mslive\react-native-hms\src\classes\HMSSDK.tsx:null in build
at node_modules\@babel\runtime\helpers\regeneratorRuntime.js:null in tryCatch
at node_modules\@babel\runtime\helpers\regeneratorRuntime.js:null in <anonymous>
at node_modules\@babel\runtime\helpers\regeneratorRuntime.js:null in tryCatch
at node_modules\@babel\runtime\helpers\regeneratorRuntime.js:null in invoke
at node_modules\@babel\runtime\helpers\regeneratorRuntime.js:null in PromiseImpl$argument_0
at node_modules\promise\setimmediate\core.js:null in tryCallTwo
at node_modules\promise\setimmediate\core.js:null in doResolve
at node_modules\promise\setimmediate\core.js:null in Promise
at node_modules\@babel\runtime\helpers\regeneratorRuntime.js:null in callInvokeWithMethodAndArg
at node_modules\@babel\runtime\helpers\regeneratorRuntime.js:null in _invoke
at node_modules\@babel\runtime\helpers\regeneratorRuntime.js:null in exports.async
at node_modules\@100mslive\react-native-hms\src\classes\HMSSDK.tsx:null in build
at App.tsx:null in useEffect$argument_0
at node_modules\react-native\Libraries\Renderer\implementations\ReactNativeRenderer-dev.js:null in commitHookEffectListMount
at node_modules\react-native\Libraries\Renderer\implementations\ReactNativeRenderer-dev.js:null in commitPassiveMountOnFiber
at node_modules\react-native\Libraries\Renderer\implementations\ReactNativeRenderer-dev.js:null in commitPassiveMountEffects_complete
at node_modules\react-native\Libraries\Renderer\implementations\ReactNativeRenderer-dev.js:null in commitPassiveMountEffects_begin
at node_modules\react-native\Libraries\Renderer\implementations\ReactNativeRenderer-dev.js:null in commitPassiveMountEffects
at node_modules\react-native\Libraries\Renderer\implementations\ReactNativeRenderer-dev.js:null in flushPassiveEffectsImpl
at node_modules\react-native\Libraries\Renderer\implementations\ReactNativeRenderer-dev.js:null in flushPassiveEffects
at node_modules\react-native\Libraries\Renderer\implementations\ReactNativeRenderer-dev.js:null in performSyncWorkOnRootat node_modules\react-native\Libraries\Renderer\implementations\ReactNativeRenderer-dev.js:null in flushSyncCallbacks
at node_modules\react-native\Libraries\Renderer\implementations\ReactNativeRenderer-dev.js:null in flushSyncCallbacksOnlyInLegacyMode
at node_modules\react-native\Libraries\Renderer\implementations\ReactNativeRenderer-dev.js:null in scheduleUpdateOnFiberat node_modules\react-native\Libraries\Renderer\implementations\ReactNativeRenderer-dev.js:null in updateContainer
at node_modules\react-native\Libraries\Renderer\implementations\ReactNativeRenderer-dev.js:null in render
at node_modules\react-native\Libraries\ReactNative\renderApplication.js:null in renderApplication
at node_modules\react-native\Libraries\ReactNative\AppRegistry.js:null in runnables.appKey.run
at node_modules\react-native\Libraries\ReactNative\AppRegistry.js:null in runApplication
at node_modules\react-native\Libraries\BatchedBridge\MessageQueue.js:null in __callFunction
at node_modules\react-native\Libraries\BatchedBridge\MessageQueue.js:null in __guard$argument_0
at node_modules\react-native\Libraries\BatchedBridge\MessageQueue.js:null in __guard
at node_modules\react-native\Libraries\BatchedBridge\MessageQueue.js:null in callFunctionReturnFlushedQueue`

react-native-cli: 2.0.1
react-native: 0.68.2
expo: 6.0.0

By making an adjust to HMSSDK.tsx, I am able to get rid of the error. I change line 92 from 

```
let id = await HMSManager.build(params?.trackSettings || {});
```

to 

```
let id = "";
if (params != undefined){
  id = await HMSManager.build(params.trackSettings);
}
```

**Comment by charles-mersereau** [comment]
I've tried developing further and learned that the adjustment I made above does not actually work. I'm new to React Native and 100ms so don't know all the ins and outs. I see now that in HMSSDK.tsx on line 92, the function `HMSManager.build()` needs to be called and cannot be skipped as that manager is needed to perform other function calls to the SDK. The problem is that any time it is called without parameters, it throws a null error. I don't know anything about HMSTrackSettings to know how to create an object to give to it.
